### ISSUE TITLE
Learning rate used for fine-tuning on ImageNet-1k

### ISSUE THREAD

**Issue by MohammedSB** [facebookresearch/dinov2#252]
With regards to this sentence:

> In Table 5, we show that the Top-1 accuracy on the validation set of ImageNet-1k improves by more than +2% when the backbone is fine-tuned.

What was the learning rate used for finetuning the backbone?

**Comment by barbolo** [comment]
I'm also looking for this, but I guess I'll not find the exact learning rate that was used.

In the [evaluation code](https://github.com/facebookresearch/dinov2/blob/main/dinov2/eval/linear.py#L145) you can find all learning rates they used in order to choose the best:

In the [paper](https://arxiv.org/abs/2304.07193), you can find this:

<img width="654" alt="image" src="https://github.com/facebookresearch/dinov2/assets/274526/4e15987a-f876-450f-a705-3e87bc0be13d">

It would be awesome if the results of the grid search are published somewhere. I'm still looking for them.

**Comment by MohammedSB** [comment]
These learning rate values are for linear probing, not fine-tuning. 
For fine-tuning, what worked for me was doing a grid search for values {1e-5, 1e-6, 1e-7}, and lower learning rates worked better for larger models.

But I used "standard" fine-tuning recipe as opposed to the one in Beit/Deit, which DINOv2 uses, so you should probably use that one if you are just starting.

**Comment by amundra15** [comment]
@MohammedSB  can you explain what you mean by "standard" fine-tuning?

**Comment by MohammedSB** [comment]
I just mean that I did not use layer-wise LR scaling, weight decay, LR warmup, mixup, or any of the other techniques in Deit III fine-tuning recipe, which DINOv2 uses.
https://github.com/facebookresearch/deit/blob/main/main.py

I basically just used an LR decay scheduler and that is it.

**Comment by amundra15** [comment]
Understood.
What was the LR that worked best for you? Also, any chance you tried freezing the initial layers and only fine-tuning the later ones?

**Comment by MohammedSB** [comment]
You can scan these values {5e-5 1e-5, 5e-6, 1e-6, 1e-7}.
Like I said, a lower learning rate worked better for me for larger models.
The optimal LR changed from dataset to dataset.

I didn't try freezing initial layers, though I suspect it won't bring much benefit. Honestly, if you want to fine-tune the model just use the deit codebase I shared, it is really strong ViT fine-tuning recipe.

**Comment by tian1327** [comment]
Hi all, I run standard finetuning or end-to-end finetuning (i.e. adding a linear classifier atop DINOv2 backbone) and achieved 5% accuracy improvement when finetuning on 16-shot in Semi-Aves dataset. I used the training strategy in the paper [SWAT](https://tian1327.github.io/SWAT/), which uses a larger learning rate of 1e-4 for classifier and a smaller learning rate of 1e-6 for the backbone. This learning rate setting shows to preserve the pretrained features and helps stabilize finetuning.

**Comment by PMRS-lab** [comment]
May I ask how to obtain the information mentioned by the author <ROOT>/labels.txt file? Thank you very much!

**Comment by tengshaofeng** [comment]
> tian1327

Great to hear that. You really did a great job. Can you show your code for finetuning dinov2 to me? Thanks.
